File: 백준/Gold/2447. 별 찍기 － 10/별 찍기 － 10.c
#include <stdio.h>
int stars[2188][2188];
int input;
void blank(int n){
    for (int i = 0; i < input; i++) {
        for (int j = 0; j < input; j++) {
            if ((n/3 <= i%n && i%n < (2*n)/3) && (n/3 <= j%n && j%n < (2*n)/3)) {
                stars[i][j] = 1;
            }
        }
    }
    
    if (n > 3) {
        blank(n / 3);
    }
}

int main(){
    scanf("%d", &input);
        
    blank(input);
    
    for (int i = 0; i < input; i++) {
        for (int j = 0; j < input; j++) {
            if (stars[i][j] == 1) {
                printf(" ");
            }
            else{
                printf("*");
            }
        }
        printf("\n");
    }
    return 0;
}

File: 백준/Gold/2447. 별 찍기 － 10/별 찍기 － 10.py
import sys
from typing import *

def erase_center(n: int) ->  None:
    global N, stars 
    for i in range(N):
        for j in range(N):
            if (n // 3 <= int(i % n) < (2 * n) // 3) and \
                (n // 3 <= int(j % n) < (2 * n) // 3):
                stars[i][j] = ' '
    
    if (n > 3):
        erase_center(n / 3)

if __name__ == '__main__':
    N: int = int(input())
    stars: List[List[str]] = [['*' for _ in range(N)] for _ in range(N)]

    erase_center(N)
    for i in range(N):
        print("".join(stars[i]))
